src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -712,11 +712,49 @@ private BoundExpression BindDeconstructionDeclarationLocal(TypeSyntax typeSyntax
                     Error(diagnostics, ErrorCode.ERR_DeconstructionVarFormDisallowsSpecificType, designation);
```

Comment:
I have left the check here. If we want to report it earlier, then it will have to be in two places...
Added test for that though